### PR TITLE
Adding new version of BH producer, changing electron eta from 2.4 to 2.5 and fixing the keep and drop list

### PR DESCRIPTION
**Pull request by adeiorio**

### CODE REVIEW

File: modules/BHProducer.py [hunk 1]
```diff
@@ -457,64 +429,55 @@ def analyze(self, event):
...
-            tau_v4_temp.SetPtEtaPhiM(
-                taus[itau].pt, taus[itau].eta, taus[itau].phi, taus[itau].mass)
+        for tau in taus:
             pass_tau_lep_Dr = 1
-            if taus[itau].pt > 20 and abs(taus[itau].eta) < 2.3 and taus[itau].idDecayModeOldDMs and taus[itau].idDeepTau2017v2p1VSe >= 4 and taus[itau].idDeepTau2017v2p1VSjet >= 4 and taus[itau].idDeepTau2017v2p1VSmu >= 1:
-                #      if taus[itau].pt>20 and abs(taus[itau].eta)<2.3 and taus[itau].idDecayModeNewDMs and taus[itau].idDeepTau2017v2p1VSe>=4 and taus[itau].idDeepTau2017v2p1VSjet>=4 and taus[itau].idDeepTau2017v2p1VSmu>=1: # use this for non-nanoaodv9
-                for ilep in range(0, len(tightLeptons)):
-                    if tau_v4_temp.DeltaR(tightLeptons[ilep]) < 0.4:
+            #if tau.pt > 20 and abs(tau.eta) < 2.3 and tau.idDecayModeOldDMs and tau.idDeepTau2017v2p1VSe >= 4 and tau.idDeepTau2017v2p1VSjet >= 4 and tau.idDeepTau2017v2p1VSmu >= 1:
+            if tau.pt>20 and abs(tau.eta)<2.3 and abs(tau.dz)<0.2 and tau.idDecayModeNewDMs and tau.idDeepTau2017v2p1VSe>=4 and tau.idDeepTau2017v2p1VSjet>=4 and tau.idDeepTau2017v2p1VSmu>=1: # use this for non-nanoaodv9
```
Comment (gourangakole) [inline modules/BHProducer.py:474]: Since we already swich to newDM here , just few DM need to reject as twiki says 
Tagging
@adeiorio @tihsu99

Comment (gourangakole) [reply modules/BHProducer.py]: Look good. Thanks a lot

**Comment by gourangakole** [comment]
@tihsu99 

Can you please check once:
Here is direct link for the producer at latest:
https://github.com/adeiorio/ttc_bar/blob/0073c473562a088baecd928fa9220f05a55e554c/modules/BHProducer.py

**Comment by tihsu99** [comment]
@adeiorio Thanks. @gourangakole if you don't have further comment, maybe we can merge it?

**Comment by gourangakole** [comment]
I will once again (in crab)

FYI:

in a private chat @adeiorio shows that there was some jobs are continuously failing in crab errors are not obvious.

here is the link:
https://monit-grafana.cern.ch/d/cmsTMDetail/cms-task-monitoring-task-view?orgId=11&var-user=adeiorio&var-task=240607_101429%3Aadeiorio_crab_SingleMuon_B&from=1717751669000&to=now

 Once the test is successfull I shall merge. 
 
 In the meantime 
 
 you can use his branch:
 
 
 ```
 git checkout -b adeiorio-lxplus-9 lxplus-9
git pull git@github.com:adeiorio/ttc_bar.git lxplus-9
```

File: modules/BHProducer.py (hunk 1 above)
Comment (gourangakole) [reply modules/BHProducer.py]: But now I check the nanoAOD docs:
there is no ``newDecayMode`` so lets stick to ``Tau_idDecayModeOldDMs``
https://twiki.cern.ch/twiki/bin/viewauth/CMS/TauIDRecommendationForRun2#Tau_Identification

but other deep id is ok. I think just the decay mode is old vs new.

**Comment by tihsu99** [comment]
I just noticed when performing object review for other group. For btagging, the suitable eta range could be up to 2.5, right?(Due to tracker boundary, 2.4 should correspond to CSC boundary) Do you know why our jet range is 2.4?

**Pull request merged**